src/OrleansRuntime/Silo/Silo.cs

```diff
@@ -179,7 +179,13 @@ internal Silo(SiloInitializationParameters initializationParams, IServiceProvide
                 serviceCollection.AddSingleton<Silo>(this);
                 serviceCollection.AddSingleton(initializationParams);
                 serviceCollection.AddLegacyClusterConfigurationSupport(config);
-                DefaultSiloServices.AddDefaultServices(null, serviceCollection);
+                var hostContext = new HostBuilderContext(new Dictionary<object, object>());
+                DefaultSiloServices.AddDefaultServices(hostContext, serviceCollection);
+
+                var parts = hostContext.GetApplicationPartManager();
+                parts.AddApplicationPartsFromDefaultDirectories();
```

Comment:
We discussed that we're not preserving the "old behavior" by processing assemblies from file system and appdomain, so this should be removed, not?

Reply:
how could it possibly work if we're not doing this here? This code is executed when a user constructs a Silo instance directly (no builder), eg when using the Azure host. This is a legacy support mechanism.

Whether or not we stop supporting that and remove this block is a matter for another PR